de.metas.business/src/main/java/de/metas/payment/api/IPaymentDAO.java

```diff
@@ -113,4 +99,6 @@
 	 * Updates the discount and the payment based on DateTrx and the payment term policy.
 	 */
 	void updateDiscountAndPayment(I_C_Payment payment, int c_Invoice_ID, I_C_DocType c_DocType);
+
+	ImmutableSet<PaymentId> retrieveAllMatchingPayments(boolean isReceipt, @NonNull BigDecimal paymentAmount, @NonNull CurrencyId currencyId, @NonNull BPartnerId bPartnerId);
```

Comment:
use `Money` instead of `BigDecimal paymentAmount`  and `CurrencyId currencyId,`

Reply:
done